File: docs/legacy(old PACE)/unsupported/VB.NET/PaceWebServiceSDK/sample/CreateEstimate.vb
Imports PaceWebServiceSDK.efipaceservices

Public Class CreateEstimate
    Public Shared Sub Run()
        Dim invokeActionHttpBinding As InvokeActionHttpBinding = PaceClient.getInvokeActionHttpBinding()
        Dim estimateInfo As EstimateInfo = createEstimateInfo()
        estimateInfo.estimatePartInfo = CreateEstimatePart.createEstimatePartInfo("")
        Try
            Dim estimate As Estimate = invokeActionHttpBinding.createEstimate(estimateInfo)
            Console.WriteLine("Estimate " + estimate.estimateNumber + " was successfully created for the customer HOUSE")
        Catch ex As Exception
            Console.WriteLine(ex.Message)
        End Try


    End Sub

    Public Shared Function createEstimateInfo() As EstimateInfo
        Dim estimateInfo As EstimateInfo = New EstimateInfo()
        estimateInfo.customer = "HOUSE"
        estimateInfo.estimateDescription = "Test Description"
        Return estimateInfo
    End Function

End Class

File: docs/legacy(old PACE)/C#/4.0/PaceAPISample/sample/CreateEstimate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using efipaceservices;
using System.Web.Services.Protocols;

namespace Pace_Web_Service_SDK.sample
{
    public class CreateEstimate
    {
        public static void Run()
        {
            InvokeActionHttpBinding invokeActionHttpBinding = PaceClient.getInvokeActionHttpBinding();
            EstimateInfo estimateInfo = createEstimateInfo();
            estimateInfo.estimatePartInfo = CreateEstimatePart.createEstimatePartInfo(null);
            try
            {
                Estimate estimate = invokeActionHttpBinding.createEstimate(estimateInfo);
                Console.WriteLine( "Estimate " + estimate.estimateNumber  + " was successfully created for the customer HOUSE");
            }
            catch (SoapException e)
            {
                Console.WriteLine(e.Message);
            }
            
        }

        private static EstimateInfo createEstimateInfo()
        {
            EstimateInfo estimateInfo = new EstimateInfo();
            estimateInfo.customer = "HOUSE";
            estimateInfo.estimateDescription = "TEST Description";
            return estimateInfo;
        }
    }

}
